modules/network/src/main/java/org/apache/ignite/network/Network.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Entry point for network module.
+ */
+public class Network {
```

Comment:
As I can see responsibilities of this class, it mostly manages workflow stuff. Also we already have an abstraction that meant to be a single entry point into networking module: `NetworkCluster`.

I suggest to merge these two entities with simultaneous splitting their set of responsibilities into three parts: workflow-related (start/shutdown/register messages from other components), discovery-related (local member, all members, events handling) and messaging-related (send, sendNoAck, sendWithResponse).

But it should be done in another task.

Reply:
Well, this already looks close that what you called "workflow-related" entity, we just need to cleanup NetworkCluster a little bit, right?